File: Lista/quest11/classesabstratas.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas_trabalhadas, valor_hora):
        self.horas_trabalhadas = horas_trabalhadas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas_trabalhadas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_mensal):
        self.salario_mensal = salario_mensal

    def calcular_salario(self):
        return self.salario_mensal

if __name__ == "__main__":
    horista = FuncionarioHorista(40, 15.00)
    assalariado = FuncionarioAssalariado(3000.00)

    print(f"Salário do Funcionário Horista: {horista.calcular_salario():.2f}")
    print(f"Salário do Funcionário Assalariado: {assalariado.calcular_salario():.2f}")

File: Lista/quest11/classesabstratas.java
abstract class Funcionario {
    public abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private double horasTrabalhadas;
    private double valorHora;

    public FuncionarioHorista(double horasTrabalhadas, double valorHora) {
        this.horasTrabalhadas = horasTrabalhadas;
        this.valorHora = valorHora;
    }

    @Override
    public double calcularSalario() {
        return horasTrabalhadas * valorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salarioMensal;

    public FuncionarioAssalariado(double salarioMensal) {
        this.salarioMensal = salarioMensal;
    }

    @Override
    public double calcularSalario() {
        return salarioMensal;
    }
}

public class Main {
    public static void main(String[] args) {
        Funcionario horista = new FuncionarioHorista(40, 15.00);
        Funcionario assalariado = new FuncionarioAssalariado(3000.00);

        System.out.printf("Salário do Funcionário Horista: %.2f\n", horista.calcularSalario());
        System.out.printf("Salário do Funcionário Assalariado: %.2f\n", assalariado.calcularSalario());
    }
}
